File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        nums = sorted(list(set(nums)))
        count = 1
        temp = 1
        for i in range(len(nums) - 1):
            if nums[i] + 1 == nums[i+1]:
                temp += 1
            else:
                count = max(count, temp)
                temp = 1

        count = max(count, temp)
        return count

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-5.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char, int> index_map;
        int i = 0, maxLength = 0;

        for (int j = 0; j < s.size(); j++) {
            if (index_map.count(s[j])) {
                i = max(i, index_map[s[j]] + 1);
            }
            index_map[s[j]] = j;
            maxLength = max(maxLength, j - i + 1);
        }
        return maxLength;
    }
};